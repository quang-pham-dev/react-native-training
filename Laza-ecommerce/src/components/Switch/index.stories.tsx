import React from 'react';
import { storiesOf } from '@storybook/react-native';
import CenterView from '../../../storybook/stories/CenterView';
import Switch from './index';

storiesOf('Switch', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Default', () => <Switch />)
  .add('With False Value', () => <Switch value={false} />);
