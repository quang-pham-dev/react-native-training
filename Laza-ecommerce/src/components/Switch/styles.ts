import { Platform, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  switch: {
    transform:
      Platform.OS === 'android'
        ? [{ scaleX: 0.9 }, { scaleY: 0.9 }]
        : [{ scaleX: 0.6 }, { scaleY: 0.6 }]
  }
});

export default styles;
