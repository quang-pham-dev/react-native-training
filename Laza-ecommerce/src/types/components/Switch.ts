export interface ISwitchProps {
  type?: 'android' | 'ios';

  value?: boolean;

  disable?: boolean;

  onToggle?: (newValue: boolean) => void;
}
