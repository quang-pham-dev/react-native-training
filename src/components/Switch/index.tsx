import React, {useCallback} from 'react'

// Libs
import {Switch as SwitchLIB} from 'react-native-switch'

// Themes
import {Colors} from '@themes'

type SwitchProps = {
  value?: boolean
  disabled?: boolean
}

const Switch = ({value = true, disabled = false}: SwitchProps) => {
  const [isEnabled, setIsEnabled] = React.useState(value)

  const handleToggle = useCallback((event: boolean) => {
    setIsEnabled(event)
  }, [])

  return (
    <SwitchLIB
      value={isEnabled}
      onValueChange={handleToggle}
      circleSize={16}
      barHeight={18}
      disabled={disabled}
      circleBorderWidth={0}
      backgroundActive={Colors.palette.emerald}
      backgroundInactive={Colors.palette.alto}
      circleActiveColor={Colors.palette.white}
      circleInActiveColor={Colors.palette.white}
      changeValueImmediately={true}
      renderActiveText={false}
      renderInActiveText={false}
      switchLeftPx={2}
      switchRightPx={2}
      switchWidthMultiplier={2}
      switchBorderRadius={30}
    />
  )
}

export default Switch
