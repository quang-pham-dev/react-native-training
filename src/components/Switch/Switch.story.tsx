import * as React from 'react'

// Storybook
import {storiesOf} from '@storybook/react-native'
import {StoryScreen, Story, UseCase} from '../../../storybook/views'

// Components
import Switch from './index'

declare let module: NodeModule

storiesOf('Switch', module)
  .addDecorator(fn => <StoryScreen>{fn()}</StoryScreen>)
  .add('Default', () => (
    <Story>
      <UseCase text="Default">
        <Switch />
      </UseCase>
    </Story>
  ))
  .add('Disabled', () => (
    <Story>
      <UseCase text="Disabled">
        <Switch disabled value={false} />
      </UseCase>
    </Story>
  ))
