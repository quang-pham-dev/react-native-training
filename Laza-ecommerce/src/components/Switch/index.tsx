import React, { useCallback } from 'react';
import { View, Switch as RNSwitch } from 'react-native';
import Colors from 'themes/Colors';

// Styles
import styles from './styles';

interface ISwitchProps {
  type?: 'android' | 'ios';

  value?: boolean;

  disable?: boolean;

  onToggle?: (newValue: boolean) => void;
}

const Switch = ({ onToggle, value = true }: ISwitchProps) => {
  const [isEnabled, setIsEnabled] = React.useState(value);

  const handleToggle = useCallback((event: boolean) => {
    setIsEnabled(event);
  }, []);

  const trackColor = {
    false: Colors.lightGray,
    true: Colors.lightGreen
  };

  return (
    <View>
      <RNSwitch
        style={styles.switch}
        trackColor={trackColor}
        thumbColor={isEnabled ? Colors.white : Colors.lightGray}
        ios_backgroundColor={Colors.lightGray}
        onValueChange={handleToggle}
        value={isEnabled}
      />
    </View>
  );
};

export default Switch;
