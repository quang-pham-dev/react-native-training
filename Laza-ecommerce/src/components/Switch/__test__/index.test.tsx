import React from 'react';

// LIBS
import renderer from 'react-test-renderer';

// Components
import Switch from 'components/Switch';

describe('Switch Component', () => {
  const tree = renderer.create(<Switch />);
  test('Should render correctly', () => {
    const component = tree.toJSON();
    expect(component).toMatchSnapshot();
  });
});
