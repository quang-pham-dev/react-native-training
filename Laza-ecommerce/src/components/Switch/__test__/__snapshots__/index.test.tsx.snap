// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar Component Should render correctly 1`] = `
<View>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#34C759"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Array [
          Object {
            "transform": Array [
              Object {
                "scaleX": 0.6,
              },
              Object {
                "scaleY": 0.6,
              },
            ],
          },
          Object {
            "backgroundColor": "#F2F2F2",
            "borderRadius": 16,
          },
        ],
      ]
    }
    thumbTintColor="#fff"
    tintColor="#F2F2F2"
    value={true}
  />
</View>
`;
